internal/workload/podspec_updates.go

```diff
@@ -47,12 +48,18 @@ const (
 
 var l = logf.Log.WithName("internal.workload")
 
-// PodAnnotation returns the key and value for an annotation on a pod
-// to indicate the specific AuthProxyWorkload resource that was configured
-// on the pod.
+// PodAnnotation returns the annotation (key, value) that should be added to
+// pods that are configured with this AuthProxyWorkload resource. This takes
+// into account whether the AuthProxyWorkload exists or was recently deleted.
```

Comment:
Should we have some unit tests around this since it's an exported function and used elsewhere?

Reply:
Yes. Unit tests added for deleted proxy in podspec_updates_test.go